src/Sarif.UnitTests/Writers/SarifLoggerTests.cs

```diff
@@ -420,5 +404,161 @@ public void SarifLogger_ResultAndRuleIdMismatch()
                 sarifLogger.Log(rule, result);
             }
         }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_ComputeFileHashes()
+        {
+            TestForLoggingOption(LoggingOptions.ComputeFileHashes);
+        }
+
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_None()
+        {
+            TestForLoggingOption(LoggingOptions.None);
+        }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_PersistEnvironment()
+        {
+            TestForLoggingOption(LoggingOptions.PersistEnvironment);
+        }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_PersistFileContents()
+        {
+            TestForLoggingOption(LoggingOptions.PersistFileContents);
+        }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_PrettyPrint()
+        {
+            TestForLoggingOption(LoggingOptions.PrettyPrint);
+        }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_Verbose()
+        {
+            TestForLoggingOption(LoggingOptions.Verbose);
+        }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_All()
+        {
+            TestForLoggingOption(LoggingOptions.All);
+        }
+
+        [TestMethod]
+        public void SarifLogger_LoggingOptions_Count()
+        {
+            // This test exists in order to alert test developers when a new member is added to the
+            // LoggingOptions enum. In that case, this test and others should be updated/added
+            // to account for the new member.
+            //     
+            // Current values are:
+            // None, ComputeFileHashes, PersistEnvironment, PersistFileContents, PrettyPrint, Verbose, All  
+            Enum.GetNames(typeof(LoggingOptions)).Length.Should().Be(7);
+        }
+
+        private void TestForLoggingOption(LoggingOptions loggingOption)
```

Comment:
>loggingOption [](start = 57, length = 13)

`loggingOptions`

Reply:
not to be pedantic but this helper is intended to test a single defined option only, not arbitrary combinations. so this name was consciously proposed

---
In reply to: [135838471](https://github.com/Microsoft/sarif-sdk/pull/609#discussion_r135838471) [](ancestors = 135838471)